src/array/growable/mod.rs

```diff
@@ -43,6 +43,10 @@ pub trait Growable<'a> {
     /// Extends this [`Growable`] with null elements, disregarding the bound arrays
     fn extend_validity(&mut self, additional: usize);
 
+    /// Get the offset of the next value to be added to this [`Growable`].
+    /// Used to remap offsets when building up a [`GrowableUnion`].
+    fn next_offset(&self) -> usize;
```

Comment:
I wonder - isn't this always equal to the logical `length` of the current stuff being built? If yes, maybe we could rename it to `len` (and therefore make it so that it is not really specific to build a `GrowableUnion` but useful to more use-cases?)

Reply:
I think so -- I was less confident that was going to be the fact when I started the implementation, but agree that seems to be the case at this point. Will rename to `len()`

Reply:
The added benefit of naming this length is logically testing it in the other unit-tests independently of union-specific offsets so we can get code-coverage on it without needing to make unions of each sub-type.